### ISSUE TITLE
[Bug]: Starting with 1.41 expect_response method can't match the simplest glob patterns

### ISSUE THREAD

**Issue by ls-alex-rzhechytskyi** [microsoft/playwright-python#2528]
### Version

1.46.0

### Steps to reproduce

Make a call using the expect_response method with a glob pattern.
e.g.
with page.expect_response("https://\*\*example.com/\*\*"):
page. get_by_text("trigger response").click()

### Expected behavior

A successful match is expected

### Actual behavior

The Playwright is not matching the URL with the pattern.
As a workaround, replaced all glob patterns with regular expressions.

### Additional context

The issue was introduced in 1.41 as it works correctly with 1.40 and remains in all versions up to 1.46

### Environment

```Text
- Operating System: [macOS 14.6.1 (23G93)]
- CPU: [Apple M2 Pro]
- Browser: [Chromium, Firefox]
- Python Version: [3.11.3]
- Other info:
```

**Comment by mxschmitt** [comment]
Would it be possible to provide a full reproduction which we can run locally?

**Comment by ls-alex-rzhechytskyi** [comment]
Hello, I'm afraid I can't do it for you right now and hopefully, it won't lead to the ticket decline. 
It's really easily reproducible using wildcards in the middle of the pattern for any expect_response call. From what I can see, some undocumented changes(I could not find any related changes in the release notes) were made to the pattern-to-regex converter somewhere within URLMatcher class.
Here's some real-life info:
url_pattern_used_for_calls = 'https://**admin.views.options**'
pw_converted_pattern_1.40 = '(?s:https://(?>.*?admin\\.views\\.options).*)\\Z'
pw_converted_pattern_1.41+ = '^https://([^/]*)admin\\.views\\.options([^/]*)$'

**Comment by ls-alex-rzhechytskyi** [comment]
The issue extends to all the methods that use the URLMatcher class (quite obvious:)).

**Comment by mxschmitt** [comment]
Hi! While this behaviour has changed in v1.41.0, we consider it a bugfix, since we aligned it with how the other languages are doing it. Looking at https://github.com/microsoft/playwright-python/issues/2193. The difference is that our glob `**` does only match until the `/` while before Playwright for Python's glob was different to what other Playwright language ports were doing - which wasn't great either. I apologise the churn - we haven't seen much negative feedback around that area, since the glob implementation is mostly the same except this.

tl;dr: To fix it, you need to do `**/*` if you want to match across the `/`: `https://**/*admin.views.options**`

**Comment by ls-alex-rzhechytskyi** [comment]
That should do the trick, appreciate a lot.
